src/tests/test_magic.py

```diff
@@ -1987,3 +1987,100 @@ def test_accessing_previously_nonexisting_file(ip_empty, tmp_empty, capsys):
     ip_empty.run_cell("%sql SELECT * FROM 'data.csv' LIMIT 3")
     out, _ = capsys.readouterr()
     assert expected in out
+
+
+def test_warn_when_using_snippets_in_non_SELECT_command(ip_empty, capsys):
+    ip_empty.run_cell("%sql duckdb://")
+    ip_empty.run_cell("%sql create table languages (name VARCHAR, rating INTEGER)")
+    ip_empty.run_cell(
+        """%%sql
+INSERT INTO languages VALUES ('Python', 1), ('Java', 0), ('OCaml', 2)"""
+    )
+
+    ip_empty.run_cell(
+        """%%sql --save language_lt1
+select * from languages where rating < 1"""
+    )
+
+    ip_empty.run_cell(
+        """%%sql --save language_lt2
+select * from languages where rating < 2"""
+    )
+
+    ip_empty.run_cell(
+        """%%sql --save language_lt3
+select * from languages where rating < 3"""
+    )
+
+    with pytest.raises(UsageError) as _:
+        ip_empty.run_cell(
+            """%%sql
+create table langs as (
+    select * from language_lt1
+)
+"""
+        )
+
+    with pytest.raises(UsageError) as _:
+        ip_empty.run_cell(
+            """%%sql
+with langs as (
+    select * from language_lt2
+) select * from langs
+"""
+        )
+
+    ip_empty.run_cell(
+        """%%sql
+create table langs1 as (
+    WITH language_lt3 as (
+        select * from languages where rating < 3
+    )
+    select * from language_lt3
+)
+"""
+    )
+
+    out, _ = capsys.readouterr()
+    print(out)
+    assert (
+        """Your query is using the following snippets: language_lt1. \
+The query is not a SELECT type query and as snippets only work \
+with SELECT queries, CTE generation is disabled"""
+        in out
+    )
+    assert (
+        """Your query is using the following snippets: language_lt2. \
+JupySQL does not support snippet expansion within CTEs yet, \
+CTE generation is disabled"""
+        in out
+    )
+    assert (
+        """Your query is using the following snippets: language_lt3. \
+The query is not a SELECT type query and as snippets only work \
+with SELECT queries, CTE generation is disabled"""
+        in out
+    )
+
+
+def test_get_query_type():
+    assert (
```

Comment:
also, add one test when calling this with None or invalid SQL, in such case we should return None

Reply:
[Fixed](https://github.com/ploomber/jupysql/pull/878/files#diff-d327143929530bcece2cae65d59465708cdf87f499fd83b4bd939c332150f713R2077-R2084)